src/ml_gym/persistency/io.py

```diff
@@ -81,23 +175,112 @@ def add_config_string(self, grid_search_id: str, config_name: str, config: str,
         return GridSearchRestfulAPIClient._set_raw_text_file_resource(url, payload_dict)
 
     def get_validation_config(self, grid_search_id: str):
+        """
+        ``HTTP GET Call Request`` Fecth validation
+          given the grid search ID over HTTP call.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+        
+        :returns: validation config json
+        """
         url = f"{self.endpoint}/{grid_search_id}/validation_config"
         return GridSearchRestfulAPIClient._get_json_resource(url)
 
     def get_experiments(self, grid_search_id: str):
+        """
+        ``HTTP GET Call Request`` Fecth experiments
+          given the grid search ID over HTTP call.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+        
+        :returns: experiments json
+        """
         url = f"{self.endpoint}/{grid_search_id}/experiments"
         return GridSearchRestfulAPIClient._get_json_resource(url)
 
     def get_full_checkpoint(self, grid_search_id: str, experiment_id: str,  checkpoint_id: int):
+        """
+        ``HTTP GET Call Request`` Fecth full checkpoint
+          given the grid search ID, experiment ID & checkpoint ID over HTTP call.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             checkpoint_id (int): Checkpoint ID
+        
+        :returns: checkpoint json
+        """
         url = f"{self.endpoint}/checkpoints/{grid_search_id}/{experiment_id}/{checkpoint_id}"
         return GridSearchRestfulAPIClient._get_json_resource(url)
 
     def get_checkpoint_resource(self, grid_search_id: str, experiment_id: str,  checkpoint_id: int,
                                 checkpoint_resource: CheckpointResource):
+        """
+        ``HTTP GET Call Request`` Fecth checkpoint resource
```

Comment:
some recurring spelling error : ""Fecth"

Reply:
perks of doing copy and paste :P . Corrected.